src/main/java/org/opensearch/security/securityconf/ConfigModelV6.java

```diff
@@ -799,69 +813,39 @@ public Set<String> getMaskedFields() {
             return Collections.unmodifiableSet(maskedFields);
         }
 
-        public Set<TypePerm> getTypePerms() {
-            return Collections.unmodifiableSet(typePerms);
-        }
-
-    }
-
-    public static class TypePerm {
-        private final WildcardMatcher typeMatcher;
-        private final Set<String> perms = new HashSet<>();
-
-        private TypePerm(String typePattern) {
-            this.typeMatcher = WildcardMatcher.ANY;
+        @Override
+        public boolean hasDlsQuery() {
+            return false;
         }
 
-        private TypePerm addPerms(Collection<String> perms) {
-            if (perms != null) {
-                this.perms.addAll(perms);
-            }
-            return this;
+        @Override
+        public boolean hasFlsFields() {
+            return false;
         }
 
         @Override
-        public int hashCode() {
-            final int prime = 31;
-            int result = 1;
-            result = prime * result + ((perms == null) ? 0 : perms.hashCode());
-            result = prime * result + ((typeMatcher == null) ? 0 : typeMatcher.hashCode());
-            return result;
+        public boolean hasMaskedFields() {
+            return false;
         }
 
-        @Override
```

Comment:
We should either keep this here (with the updated logic) or make sure it is properly inherited since this is a complex object.

Reply:
This change has been reverted